docs/UpgradeGuide.md

```diff
@@ -2,57 +2,78 @@
 
-Upgrading `ddtrace` from 0.x to 1.x introduces some changes to namespacing, the public API, and the underlying trace data structure.
+**How to upgrade basic usage**
 
-Here's a list of the most common changes you may encounter:
+For users with a basic implementation (configuration file + out-of-the-box instrumentation), only minor changes to your configuration file are required: most applications take just minutes to update. Check out the following sections for a step-by-step guide.
 
-## Namespacing & the public API
+- [Configuration](#1.0-configuration)
+  - [`require` paths have changed](#1.0-configuration-requires)
+  - [Settings have been renamed](#1.0-configuration-settings)
+  - [Activating instrumentation](#1.0-configuration-instrumentation)
+- [Instrumentation](#1.0-instrumentation)
+  - [Service naming](#1.0-instrumentation-service-naming)
 
-To avoid naming conflicts with new Datadog features and products, many of the tracing constants and functions defined directly under `Datadog` have moved to `Datadog::Tracing`.
+**Addtional upgrades for advanced usage**
 
-You can access most common functions through our [public API](https://www.rubydoc.info/gems/ddtrace/) within `Datadog::Tracing`. Use of some of the functions in the public API will be described in use cases below. Here's a few examples:
+For users with an advanced implementation of `ddtrace` (custom instrumentation, sampling or processing behavior, etc), some namespace and behavioral changes may be required. See the following sections for details about what features changed and how to use them.
 
-```ruby
-### Old 0.x ###
-Datadog.tracer.trace
-Datadog.tracer.active_span
-Datadog.tracer.active_correlation.to_s
+- [Namespacing](#1.0-namespacing)
+- [Trace API](#1.0-trace-api)
+  - [Removed `Datadog.tracer`](#1.0-trace-api-removed-tracer)
+  - [Removed access to `Datadog::Context`](#1.0-trace-api-removed-context)
+  - [Manual tracing & trace model](#1.0-trace-api-manual-tracing)
+  - [Accessing trace state](#1.0-trace-api-trace-state)
+  - [Distributed tracing](#1.0-trace-api-distributed)
+    - [Over HTTP](#1.0-trace-api-distributed-http)
+    - [Over gRPC](#1.0-trace-api-distributed-grpc)
+    - [Between threads](#1.0-trace-api-distributed-threads)
+  - [Sampling](#1.0-trace-api-sampling)
+  - [Processing pipeline](#1.0-trace-api-pipeline)
 
+**Appendix**
 
-### New 1.0 ###
-Datadog::Tracing.trace
-Datadog::Tracing.active_span
-Datadog::Tracing.log_correlation
-# ...and more...
-```
+For a comprehensive list of everything that changed, the appendix hosts some helpful and detailed tables with recommendations.
 
-Check out our [complete list of namespacing changes](#appendix-namespacing) for more details.
+  - [Namespace mappings](#1.0-appendix-namespace)
+    - [Constants](#1.0-appendix-namespace-constants)
+  - [Breaking changes](#1.0-appendix-breaking-changes)
 
-### Removed `Datadog.tracer`
+<h1 id="1.0-basic-upgrade">Upgrading basic usage</h1>
```

Comment:
It's a bit weird that while this document (`UpgradeGuide.md`) seems like it's making space for future upgrades to be included as well, most sections are using h1's, aka putting stuff on the top-level structure.

Should we make this document just the 1.x upgrade guide? In that case, it makes sense to keep the h1s. Otherwise, should everything below "From 0.x to 1.0" become an h2?